pkg/controllers/termination/suite_test.go

```diff
@@ -89,6 +89,8 @@ var _ = Describe("Termination", func() {
 			Expect(env.Client.Delete(ctx, node)).To(Succeed())
 			node = ExpectNodeExists(env.Client, node.Name)
 			ExpectReconcileSucceeded(controller, client.ObjectKeyFromObject(node))
+			// Sleep for one second to simulate queue processing work
+			time.Sleep(1 * time.Second)
```

Comment:
why do we need to sleep here?  I would try hard to not introduce sleeps into tests, it generally makes them flakey. IIUC, the test should not evict the pod, so if we continue w/o a sleep, I'm guessing you're worried that it could pass before the queue processes?  

I wonder if there's a way to synchronize the ExpectReconcileSucceeded to make sure the work queue is empty before proceeding.

Reply:
The EvictionQueue actually processes evictions independent of the reconcile loop. Because of this, I was trying to figure out how to make the program wait. Sometimes these tests would fail simply because the Reconcile loop succeeded so quickly that the EvictionQueue couldn't process the one pod in the queue and update the pod before we called GetPod in the next line.  

I played with the idea of using Eventually here, but since the way we see if a pod has successfully received an evict call is checking if the DeletionTimestamp is nonZero, propagation delay means that our Eventually would have to rely on a timeout for this to work, IIUC, which is the same as a sleep.

Reply:
I still haven't thought of a cleaner way to synchronize the queue for deterministic tests (without creating functionality in the queue that we only need for the tests), so I suppose this is okay for now...  We should definitely keep thinking on a minimalist way to get rid of this sleep though.

Reply:
The sleep is gone now!